app/controllers/UploadYourFilesReceiptController.scala

```diff
@@ -39,20 +41,29 @@ class UploadYourFilesReceiptController @Inject()(
   cdsFileUploadConnector: CdsFileUploadConnector,
   metrics: SfusMetrics,
   uploadYourFilesReceipt: upload_your_files_receipt,
-  answersConnector: AnswersService
+  uploadYourFilesConfirmation: upload_your_files_confirmation,
+  answersConnector: AnswersService,
+  secureMessagingConfig: SecureMessagingConfig
 )(implicit mcc: MessagesControllerComponents, ec: ExecutionContext)
     extends FrontendController(mcc) with I18nSupport {
 
   def onPageLoad(): Action[AnyContent] = (authenticate andThen verifiedEmail).async { implicit req =>
```

Comment:
I would refactor `onPageLoad` splitting the content in a few `def` to help the "reader" to better understand whats going on here. atm it has too much nested logic, imo. At least the `yield` block.

Reply:
Ok, I've split the code in the yield out into another private method.